File: 104-Maximum-Depth-of-Binary-Tree/solve.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        
        if (root == NULL)
            return 0;
        
        if(root->left == NULL && root->right == NULL){
            return 1;
        }
        
        int leftDepth, rightDepth;
        if(root->left == NULL)
            leftDepth = 0;
        else
            leftDepth = maxDepth(root->left);
        
        if(root->right == NULL)
            rightDepth = 0;
        else
            rightDepth = maxDepth(root->right);
        
        if (leftDepth >= rightDepth)
            return leftDepth + 1;
        else
            return rightDepth + 1; 
            
        
        
        
        
    }
};

File: 257-Binary-Tree-Paths/solve.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    List<String> res = new ArrayList<String>();
    
    public List<String> binaryTreePaths(TreeNode root) {
    
        if (root == null)
            return res;
        
        StringBuilder sb = new StringBuilder();
        getPaths(root, res, sb);
    
        return res;
        
    }
    
    public void getPaths(TreeNode root, List<String> res, StringBuilder sb){
        
        if( root.left == null && root.right == null){
            
            sb.append(root.val);
            res.add(sb.toString());
            return ;
        }
     
        sb.append(root.val);
        sb.append("->");
        int length = sb.length();
            
        if (root.left != null){
            
            getPaths(root.left, res, sb);
            
        }
        sb.setLength(length);
        
        if (root.right != null){

            getPaths(root.right, res, sb);            
        }
     
        
    }
    
        
}
